examples/transportation/segment/road/road-with-route.yaml

```diff
@@ -0,0 +1,22 @@
+---
+id: overture:transportation:segment:123
+type: Feature
+geometry:
+  type: LineString
+  coordinates: [[0, 0], [1, 1]]
+properties:
+  # Overture properties
+  theme: transportation
+  type: segment
+  update_time: "2024-04-10T00:00:00-05:00"
+  version: 1
+  subtype: road
+  class: motorway
+  connector_ids: [fooConnector, barConnector]
+  routes:
```

Comment:
I think this should go under `road:`. The idea with `road` is to contain all the complex properties.

If we think it will also apply to subtype `rail` or `water` I'd be tempted to repeat it under those subtypes' `rail:` or `water:` property.

Reply:
Exactly the same reason you said Brad. Something like name (which is also a pretty complex property with multiple different name types) which is a top level property. Another reason is that its not directly related to transportation i.e. something that impacts routing which is what most of the road properties are.

Reply:
This is what I'm working on, at a very slow pace. It's not really new, in that it is attempting to provide a coherent schema-wide description of how things (mostly) are, and how to make decisions about them.

https://wiki.overturemaps.org/x/0YIWAg#id-[Draft]SchemaStandards-NestedProperties

Reply:
@vcschapp I'm not sure benefits of nested JSON outweigh the negatives as the proposal currently stands. As I've started work with the data in depth, it has been difficult to answer some simple questions like "is foot access allowed?" in pure SQL. Here's an example query fragment I implemented to solve this:
```
JSON_ARRAY_CONTAINS(
  JSON_EXTRACT(
    TRY(JSON_PARSE(road),
    '$.restrictions.access[?(@.access_type==''allowed'')].when.mode[*]'
  ),
  'foot'
)
```
Relying on JSON parsing capabilities of query engines like this isn't ideal. Whereas if we brought something like `road.restrictions` to the top level:
```
ANY_MATCH(
  restrictions.access,
  restriction -> restriction.access_type = 'allowed' AND ANY_MATCH(restriction.when.mode, mode -> mode = 'foot')
)
```
which feels much more natural (to me 🙂) and doesn't require deep JSON path understanding. The biggest drawback to this approach (as detailed in your proposal) is an overwhelming # of columns and the potential complexity of these types for creating tables in diverse environments. Hopefully we can mitigate this with existing parquet metadata tools and better documentation.

One last thought - as our JSON object gets more complicated, the penalty to use that data in complex joins or filters grows as the full object must always be parsed and stored in memory. This means we also lose at least some of the benefit of a columnar-based storage.

Reply:
@vcschapp given the [schema discussion](https://github.com/OvertureMaps/schema-wg/issues/302#issuecomment-2088682544) today, can I consider this resolved in favor of top-level property structuring?

Reply:
Let's consider this resolved in light of the plan @brad-richardson, @ibnt1, and myself aligned on yesterday to introduce some flattening of properties currently contained under `road`.